M2P-147 Implement a feature switch to toggle the logic to pre-fill address from Bolt for the logged-in customer

johnmccombs1 opened the pull request:
# Description
There is a merchant that doesn’t want to use the logic pre-fill address from Bolt for the logged-in customer. 
So I created this PR to implement a feature switch to toggle that logic

Fixes: 
https://boltpay.atlassian.net/browse/M2P-147
https://boltpay.atlassian.net/browse/EN-395

#changelog M2P-147 Implement a feature switch to toggle the logic to pre-fill address from Bolt for the logged-in customer

# Type of change

- [ ] Bug fix (change which fixes an issue)
- [x] New feature (change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update


# How Has This Been Tested?
Please validate that you have tested your change in at least one of the following areas:

- [x] Successfully tested locally (or docker image)
- [ ] Successfully tested on a staging or sandbox server
- [ ] Successfully tested on a merchant's staging server

# For PR Reviewer 
- [ ] Reviewed unit tests to make sure we are using real components rather than mocks as much as possible?

# Checklist:

- [x] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] I have created or modified unit tests to sufficiently cover my changes.
- [x] I have added my Jira ticket link and provided a changelog message.

johnmccombs1 commented:
@danac-gs Yes, that is correct. If the feature switch is enabled, it would prevent logging in the existing Bolt accounts and force every customer to confirm their address first like this http://prntscr.com/t4ipjg

njfaries commented:
@johnmccombs1 isn't it the other way? As in, it looks like if the switch is enabled prefilling is able to happen, and if disabled then prefilling is stopped. So this switch would be on by default except this merchant (and others with similar needs) could disable? Just want to be clear because that's how it looks and what the naming seems to imply.

vitaliyreznikov commented:
`isPrefillAddressFromBoltForLoggedInCustomerEnabled` looks like we can enable or disable this as we want.
I think we need to always have this functional enable and only if we really need - disable it.
Please consider renaming feature switch into negative name, like
`ifShouldDisablePrefillAddressForLoggedInCustomer`
This feature switch can be disabled by default (rollout 0%) and we can enable it for the specific merchant.

johnmccombs1 commented:
@njfaries  I've just updated the method name so it looks more specific about the feature. By default, the feature switch is off and we allow the prefilling address for the logged-in customer. When we enable the feature switch, it won't allow the prefilling address for the logged-in customer.


@vitaliyreznikov  I've updated the code. Thanks for the suggestion
